### ISSUE TITLE
[UX] Make only one theme "active" at a time

### ISSUE THREAD

**Issue by jenlampton** [backdrop/backdrop-issues#131]
Simplification is a thing of beauty.

Let's simplify our theme system and have only one active theme at a time (except for the admin theme, everyone loves that).

When a new theme is enabled on the Appearance page, the previously enabled theme will become disabled. This will help eliminate the confusing concepts of a "default" theme as well as "global" theme settings.

**Comment by BWPanda** [comment]
While I agree with the simplification this would provide, I think @mkalkbrenner makes a good point about sub-themes. 6 of the 10 [most installed D7 themes](https://www.drupal.org/project/project_theme?f[2]=drupal_core%3A103&f[3]=sm_field_project_type%3Afull&solrsort=iss_project_release_usage+desc) are base themes. How would we account for this?

**Comment by quicksketch** [comment]
> Would a sub-theme still work effectively if its base theme was disabled?

As far as I know it already works this way. I know that every time I've used Zen I never enable the base theme. There may be an issue with themes that provide a lot of theme settings through the theme settings form, as I'm not sure forms are inherited into their sub-themes, but if they are (or we make them inherit), that seems like it would be another improvement from a UX perspective. It's confusing trying to figure out if you should set settings in the base theme or in the sub-theme.

**Comment by BWPanda** [comment]
> It's confusing trying to figure out if you should set settings in the base theme or in the sub-theme.

Absolutely! I often struggle with this...

**Comment by quicksketch** [comment]
I'd still love to see this, but we're out of time for 1.0.0. Moving this to 2.x.

**Comment by klonos** [comment]
This is the UI pattern we are using in the language list/manage page under `/admin/config/regional/language` to enable/disable languages, and to also set the default one:

<img width="701" alt="screen shot 2018-11-25 at 8 18 22 am" src="https://user-images.githubusercontent.com/2423362/48973060-2237c200-f08b-11e8-8251-b41c7628798f.png">

In #3385 we have been discussing to reuse this UI pattern in the "Search items" section in `/admin/config/search/settings`, to turn it from this:

<img width="953" alt="screen shot 2018-11-25 at 8 24 55 am" src="https://user-images.githubusercontent.com/2423362/48973088-a25e2780-f08b-11e8-8978-b830bd0a76b7.png">

...to this:

<img width="942" alt="screen shot 2018-11-25 at 8 18 52 am" src="https://user-images.githubusercontent.com/2423362/48973062-2c59c080-f08b-11e8-8dd7-2147ca5d98c7.png">

I recently span up a D6 instance, and noticed that this UI pattern was actually used for the theme listing page:

<img width="1033" alt="screen shot 2018-11-25 at 8 26 52 am" src="https://user-images.githubusercontent.com/2423362/48973100-f537df00-f08b-11e8-8b57-2c8adb26e26b.png">

**Comment by klonos** [comment]
Here's how it could look in Backdrop (mockup):

![screencapture-trusted-hosts-local-admin-appearance-2018-11-25-09_03_40](https://user-images.githubusercontent.com/2423362/48973333-354d9080-f091-11e8-8434-00b04587e0cb.png)

**Comment by klonos** [comment]
...for this specific issue here, we could ditch the "Enabled" column, and rename/repurpose the "Default" column.

**Comment by findlabnet** [comment]
> we could ditch the "Enabled" column, and rename/repurpose the "Default" column.

Maybe leave "Enabled" intact, but replace title "Default" with "Active"?

**Comment by klonos** [comment]
Here's a few points to consider:

- Are theme switching modules relevant nowadays? Has any of them been ported to Backdrop? ...in other words, should Backdrop sacrifice UI simplicity for the sake of non-popular contrib modules, which could very well implement their own config form to allow/disallow specific themes to be used.
- Disabled themes can still be used as base themes without any issues.
- The "default" theme is actually the one used in the front-end; while the backend theme is chosen via the separate drop-down in the "Admin theme" fieldset at the bottom of the form.

...so if I was to change something, it would be:

1. remove the "enabled" column
2. rename the "default" column to "frontend theme"
3. add a new "admin theme" column with radio buttons, next to the "frontend theme" column
4. remove the "admin theme" fieldset
5. keep the "use admin theme when editing..." checkbox, and perhaps move it to the top of the form, above the themes list
6. (optional) add a "use the same theme for the frontend and the admin UI"

Why 4, 5, and 6 you ask? ...a couple of reasons:

- the "admin theme" fieldset requires scrolling to get to it, and more scrolling on mobile, which makes these options less discoverable for non-Drupal/Backdrop users
- compare:
  - scroll -> click the "admin theme" drop-down -> select an option = scroll + 2 clicks 👍 
  - click one of the radio buttons in a new "admin theme" column = no scrolling + 1 click 👎

**Comment by klonos** [comment]
...having said that, this would be far off the idea of reusing a common UI pattern (which is just an "enable" column and a "default" column).

**Comment by jenlampton** [comment]
> I recently span up a D6 instance, and noticed that this UI pattern was actually used for the theme listing page

Yes, a lot of UX studies were done on the Drupal 6 themes page, and the result was the new improved appearance page in Drupal 7 (that we also have in Backdrop). I would hate to see us go backwards.

I think the main problem with this UI is that the concepts aren't intuitive to people. What is the difference between an enabled theme and a default theme? If only one theme were enabled at a time, we wouldn't need the concept of Default and everything gets much simpler. 

We can't remove the ability to have multiple themes from Backdrop core in 1.x, but since 2.x is still 5 years away, I do like the idea of moving it into a core module now, disable it by default, and when 2.x comes around we can move that module to contrib.

**Comment by indigoxela** [comment]
> When a new theme is enabled on the Appearance page, the previously enabled theme will become disabled.

I see one problem with this: if a theme gets disabled, its settings (Color...) will be lost. Personally, I wouldn't assume that when I enable another theme (when trying different ones out), the previous one looses all settings. Think of color settings that may have taken some time to set up - and then everything's gone. IMO a WTF moment.

**Comment by klonos** [comment]
> ...if a theme gets disabled, its settings (Color...) will be lost.

Not if we get things right. If we treated themes properly, like modules (see #4829), then disabling a theme should NOT be removing its settings - you'd have to uninstall the theme completely for its settings to go away.

**Comment by philsward** [comment]
Speaking of "uninstalling" themes, it would honestly be nice if modules and themes acted the same way in this regard. This always felt like a weird UX going all the way back to D5.

**Comment by laryn** [comment]
Just catching up to @klonos related issue https://github.com/backdrop/backdrop-issues/issues/4459
